File: java/jzoffer/chapter6/MaxProfit_63.java
package chapter6;

public class MaxProfit_63 {

    public int maxProfit(int[] prices) {
        int min = Integer.MAX_VALUE;
        int maxProfit = 0;
        for (int price : prices) {
            min = Math.min(min, price);
            maxProfit = Math.max(maxProfit, price - min);
        }
        return maxProfit;
    }
}

File: go/jzoffer/maxProfit_63.go
package jzoffer

import "math"

func maxProfit(prices []int) int {
	preMin := math.MaxInt32

	res := 0

	for _, price := range prices {
		if price < preMin {
			preMin = price
		}
		if res < price-preMin {
			res = price - preMin
		}
	}

	return res
}
